packages/block-library/src/template-part/edit/import-controls.js

```diff
@@ -67,6 +77,12 @@ export function TemplatePartImportControls( { area, setAttributes } ) {
 		];
 	}, [ sidebars ] );
 
+	// Only bailout from rendering after data is loaded, and there is nothing to import,
+	// to avoid SlotFill re-positioning issue. See: https://github.com/WordPress/gutenberg/issues/15641.
+	if ( hasResolved && ! options.length ) {
+		return null;
+	}
```

Comment:
The only problem with this condition is that users might see the component flash and disappear the first time they open the Advanced controls. See the attached screencast.

An alternative would be to ignore the SlotFills re-positioning bug (#15641) and only render controls after loading the data.

@jasmussen, any suggestions from a design perspective?

Reply:
Thanks to @ntsekouras, I think we found a better temporary solution.

Reply:
Sorry I missed this. I trust the both of you completely, and we can always revisit if something resurfaces.